src/Framework/ClassResolver/DocBlockService/UseBlockParser.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Gacela\Framework\ClassResolver\DocBlockService;
+
+final class UseBlockParser
+{
+    public function getUseStatement(string $className, string $phpCode): string
+    {
+        if ($phpCode === '') {
+            return '';
+        }
+
+        $fullyQualifiedClassName = $this->searchInUsesStatements($className, $phpCode);
+        if ($fullyQualifiedClassName !== '') {
+            return $fullyQualifiedClassName;
+        }
+
+        $namespace = $this->lookInCurrentNamespace($phpCode);
+
+        return sprintf('%s\\%s', $namespace, $className);
+    }
+
+    private function searchInUsesStatements(string $className, string $phpCode): string
+    {
+        $needle = "{$className};";
+        $lines = array_filter(
+            explode(PHP_EOL, $phpCode),
+            static fn (string $l) => str_contains($l, $needle)
+        );
+        /** @psalm-suppress RedundantCast */
+        $lineSplit = (array)explode(' ', (string)reset($lines));
+
+        return rtrim($lineSplit[1] ?? '', ';');
```

Comment:
@Chemaclass maybe we should check the line starts with `use ...`?

<img width="1034" alt="image" src="https://user-images.githubusercontent.com/6381924/165517944-32322d53-8c3f-4e3d-a914-e4b201c27e49.png">